lib/lure/provider/bitbucket.go

```diff
@@ -0,0 +1,253 @@
+	"github.com/coveooss/lure/lib/lure/project"
+	"github.com/coveooss/lure/lib/lure/vcs"
+
+	"github.com/sethgrid/pester"
+)
+
+type BitBucket struct {
+	URL            string
+	apiURL         string
+	authentication vcs.Authentication
+}
+
+type pullRequestList struct {
+	Next        string        `json:"next"`
+	PullRequest []PullRequest `json:"values"`
+}
+
+type branch struct {
+	Name string `json:"name"`
+}
+
+type source struct {
+	Branch branch `json:"branch"`
+}
+
+type dest struct {
+	Branch branch `json:"branch"`
+}
+
+type user struct {
+	Uuid string `json:"uuid"`
+}
+
+func New(authentication vcs.Authentication, project project.Project) BitBucket {
+	return BitBucket{
+		URL:            "https://bitbucket.org/" + project.Owner + "/" + project.Name,
+		apiURL:         "https://api.bitbucket.org/2.0/repositories",
+		authentication: authentication,
+	}
+}
+
+func (bitbucket BitBucket) GetURL() string {
+	return bitbucket.URL
+}
+
+func (bitbucket BitBucket) GetPullRequests(username string, repoSlug string, ignoreDeclinedPRs bool) ([]PullRequest, error) {
+
+	log.Logger.Info("Retrieving pull requests")
+
+	acceptedStates := "state=OPEN"
+	if !ignoreDeclinedPRs {
+		acceptedStates += "&state=DECLINED"
+	}
+
+	bitBucketPath := fmt.Sprintf("/%s/%s/pullrequests/?%s", username, repoSlug, acceptedStates)
+
+	prRequest, _ := bitbucket.createApiRequest("GET", bitBucketPath, nil)
+	prRequest.Header.Add("Content-Type", "application/json")
+
+	var list pullRequestList
+
+	tmpList, e := bitbucket.getPRRequest(prRequest)
+	if e != nil {
+		log.Logger.Error(e.Error())
+		return nil, e
+	}
+	list.PullRequest = append(list.PullRequest, tmpList.PullRequest...)
+
+	if tmpList.Next != "" {
+
+		for tmpList.Next != "" && len(tmpList.PullRequest) != 0 {
+			queryParams, _ := url.ParseQuery(tmpList.Next)
+			nextQueryParams := prRequest.URL.Query()
+			nextQueryParams.Set("page", queryParams.Get("page"))
+			prRequest.URL.RawQuery = nextQueryParams.Encode()
+			tmpList.Next = "" //Reset
+			tmpList, e = bitbucket.getPRRequest(prRequest)
+			if e != nil {
+				log.Logger.Error(e.Error())
+				return nil, e
+			}
+			list.PullRequest = append(list.PullRequest, tmpList.PullRequest...)
+		}
+	}
+	log.Logger.Infof("Found %d PRs.", len(list.PullRequest))
+
+	return list.PullRequest, nil
+}
+
+func (bitbucket BitBucket) getDefaultReviewers(username string, repoSlug string) ([]user, error) {
+
+	bitBucketPath := fmt.Sprintf("/%s/%s/default-reviewers", username, repoSlug)
+
+	request, _ := bitbucket.createApiRequest("GET", bitBucketPath, nil)
+	request.Header.Add("Content-Type", "application/json")
+
+	client := getHTTPClient()
+	resp, err := client.Do(request)
+
+	if err != nil || resp.StatusCode < 200 || resp.StatusCode > 299 {
+		log.Logger.Error("Error getting default reviewers", client.LogString())
+		return nil, errors.New("Something went wrong getting default reviewers, got status code " + resp.Status)
+	}
+
+	type GetDefaultReviewers struct {
```

Comment:
I generally think it's better to put those kind of struct used to decode JSON in the package itself as first class citizen ( instead of inside a function), and simply lowerCase it if you don't want to export it.